src/main/java/org/jabref/JabRefGUI.java

```diff
@@ -91,7 +92,7 @@ private void openWindow(Stage mainStage) {
             }
         });
 
-        Platform.runLater(this::openDatabases);
+        executorService.execute(this::openDatabases);
```

Comment:
Did you tried if this works (i.e. if JabRef still starts and loads the databases)? That would be cool, but I fear that there might be an exception thrown as things are not initialized correctly on the javafx thread.

Reply:
Well, both lines work pretty similar, which is to open the databases at some point in the future. The only difference is, that it's no longer executed by the JavaFX thread. That prevents any GUI freeze on startup. In my opinion the GUI has to be able to start up completely empty and get the data fed in as it's loaded. So in case that you are right and the program does not start correctly anymore due to this change, then we got a clear indicator for some bad code smell somewhere else in the project. Anyway - I'll test it to see how the program behaves. Just to make sure what's going on exactly.

Reply:
You are of course completely right. It should work in principle. However, the GUI code is smelly at some points and the database handling is some of the extra smelly parts. This is why I'm a bit afraid it doesn't work. If it does, perfect and we can merge! Otherwise I would leave the current solution for now.

Reply:
Alright, I finally managed to test it and it seems working to me. I imported several (smaller) databases and restarted the program a few times. Nothing too concerning happend. Therefore I think this is good to go.

Reply:
"Nothing too concerning happend. “

Sounds convincing 😂

Reply:
If @matthiasgeiger is convinced, then I'm too!